lib/Backend/GlobOpt.cpp

```diff
@@ -5690,6 +5690,13 @@ GlobOpt::OptSrc(IR::Opnd *opnd, IR::Instr * *pInstr, Value **indirIndexValRef, I
         }
         originalPropertySym = sym->AsPropertySym();
 
+        //Don't copy-prop for cases like foo.arguments
```

Comment:
Can you check if you can use argumentsEquivBv on Globopt to solve your issue? That bit vector is there for these kind of issues only- maybe we need to expand the usage of it. Basically, right now we use the bv to disable copy-prop or arguments across inlinees